### PR TITLE
Update Salesforce EMP Connector (cometd java client 4.0.4)

### PR DESCRIPTION
**Pull request by neewob**

### CODE REVIEW

File: src/main/java/com/salesforce/emp/connector/BayeuxParameters.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ default TimeUnit keepAliveUnit() {
     default Map<String, Object> longPollingOptions() {
         Map<String, Object> options = new HashMap<>();
         options.put("maxNetworkDelay", maxNetworkDelay());
-        options.put("maxBufferSize", maxBufferSize());
+        options.put("maxMessageSize", maxBufferSize());
```
Comment (neewob) [inline src/main/java/com/salesforce/emp/connector/BayeuxParameters.java:64]: Is this right? I think so. We didn't make this change btw. If you inspect the cometd java client **3.1.4**, you'll find references to both maxBufferSize and maxMessageSize properties in ClientTransport and LongPollingTransport classes. In cometd **4.0.4**, only maxMessageSize is referenced.

Comment (pmeister) [reply src/main/java/com/salesforce/emp/connector/BayeuxParameters.java:64]: Your analysis sounds reasonable.

**Comment by pmeister** [comment]
Ship it! errr, Merge it!

**Comment by pmeister** [comment]
WAIT, doesn't this need to be merged to sl-master branch?

**Comment by neewob** [comment]
Sorry for not clarifying the branches. I had spoken to Robin about this, and he suggested that I merge sl-master into master which is what I did. So, this PR will merge into the right branch. I'm going to try your suggestion of upgrading cometd java client to latest 4.0.x version (4.0.9).

**Comment by neewob** [comment]
I will hold off merging until we resolve the jetty dependency versions in Snap tree (root pom, hadoop, salesforce).

**Pull request merged**